### PR TITLE
Everywhere: Some initial pvs-studio fixes + zero memory when using Core::get_password()

### PR DESCRIPTION
**Pull request by bgianfo**

### CODE REVIEW

File: Userland/Utilities/passwd.cpp [hunk 1]
```diff
@@ -89,6 +90,10 @@ int main(int argc, char** argv)
                 return 1;
             }
 
+            ScopeGuard current_password_zero_guard = [&] {
+                explicit_bzero(const_cast<char*>(current_password.value().characters()), current_password.value().length());
```
Comment (awesomekling) [inline Userland/Utilities/passwd.cpp:94]: This feels really gross. `StringImpl` contents are supposed to be immutable (and shareable).
Let's find a better way, perhaps we could have a simple buffer type that scrubs itself on destruction?

Comment (Petelliott) [reply Userland/Utilities/passwd.cpp:94]: Why not a flag in `StringImpl` that enables zeroing the StringImpl in the destructor?

Comment (awesomekling) [reply Userland/Utilities/passwd.cpp:94]: That solves one part of it, but `StringImpl` is still a shareable type, so one individual co-owner going out of scope doesn't mean the impl gets destroyed yet.
If someone copies the `String`, the `StringImpl` lives on.

Comment (Petelliott) [reply Userland/Utilities/passwd.cpp:94]: That's fine, because if someone copies the string, the data is in memory anyway.

Comment (bgianfo) [reply Userland/Utilities/passwd.cpp]: Ok, I created a `SecretString` class that wraps a `ByteBuffer`. It's much cleaner, let me know what you think.

**Comment by awesomekling** [comment]
:yakkiss:

**Pull request merged**